### PR TITLE
iOS - Custom navigation bar view

### PR DESCRIPTION
**Pull request by gran33**
React root view on the navigation bar

### CODE REVIEW

File: ios/RCCViewController.m [hunk 1]
```diff
@@ -469,6 +469,54 @@ -(void)setStyleOnAppearForViewController:(UIViewController*)viewController appea
...
+  navigationBarFrame.size.width = customViewWidth;
+  
+  
+  return navigationBarFrame;
+}
+
+-(CGFloat)getNavBarButtonsWidth:(NSArray*)buttons {
+  CGFloat sum = 0;
+  for (UIBarButtonItem *item in buttons) {
+    UIView *view = [item valueForKey:@"view"];
```
Comment (DanielZlotin) [inline ios/RCCViewController.m]: KVO?

File: ios/RCCViewController.m [hunk 2]
```diff
@@ -469,6 +469,54 @@ -(void)setStyleOnAppearForViewController:(UIViewController*)viewController appea
   } else {
     self.navBarHairlineImageView.hidden = NO;
   }
+  
+  NSString *navBarCustomView = self.navigatorStyle[@"navBarCustomView"];
+  if (navBarCustomView) {
+    NSLog(@"navBarCustomView:%@", navBarCustomView);
```
Comment (DanielZlotin) [inline ios/RCCViewController.m]: remove

File: ios/RCCViewController.m [hunk 3]
```diff
@@ -469,6 +469,54 @@ -(void)setStyleOnAppearForViewController:(UIViewController*)viewController appea
...
+    if ([self.view isKindOfClass:[RCTRootView class]]) {
```
Comment (DanielZlotin) [inline ios/RCCViewController.m]: un needed

Comment (gran33) [inline ios/RCCViewController.m]: Defensive coding :), it's better practice to check this is the right object before `casting`.

File: ios/RCCViewController.m (hunk 1 above)
Comment (gran33) [inline ios/RCCViewController.m]: KVO it is!

**Comment by tmaly1980** [comment]
@gran33 Is this in the docs? What about Android?

**Comment by gran33** [comment]
Docs: 
https://wix.github.io/react-native-navigation/#/styling-the-navigator?id=style-object-format

Android implementation: 
https://github.com/wix/react-native-navigation/search?l=Java&q=topBarReactView&type=&utf8=%E2%9C%93

**Pull request merged**